### ISSUE TITLE
[multiple] scipy dependence (Trac #1110)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1104]
Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt

~~DOMLauncher~~ (only two examples use scipy)
~~clsim~~
~~credo~~
~~lilliput~~
~~millipede~~
~~neutrino-generator~~
~~paraboloid~~
photonics-service (Jakob)
photospline (Jakob)
~~sim-services~~
~~wavereform~~

Migrated from https://code.icecube.wisc.edu/ticket/1110
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:57", 
    "description": "Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt\n\n~~DOMLauncher~~ (only two examples use scipy)\n~~clsim~~\n~~credo~~\n~~lilliput~~\n~~millipede~~\n~~neutrino-generator~~\n~~paraboloid~~\nphotonics-service (Jakob)\nphotospline (Jakob)\n~~sim-services~~\n~~wavereform~~\n", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067117911749", 
    "component": "combo reconstruction", 
    "summary": "[multiple] scipy dependence", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2015-08-11T20:38:52", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150811T20:52:14: 
* nega changed summary from "[credo] scipy dependence" to "[multiple] scipy dependence"
* nega changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> PROPOSAL
> clsim
> credo
> lilliput
> millipede
> neutrino-generator
> paraboloid
> photonics-service
> photospline
> sim-services
> wavereform
>

**Comment by icecube-trac** [comment]
Trac update at 20150811T20:58:16: nega commented:

> See the commit message for r136085/IceCube

**Comment by icecube-trac** [comment]
Trac update at 20150811T20:59:59: 
* nega changed owner from "jtatar" to "olivas"
* nega changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20150811T21:00:44: olivas changed owner from "olivas" to "olivas, jtatar, cweaver, sflis, tfuchs,  claudio.kopper, hoshina, jbraun, jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20150812T23:55:34: 
* olivas changed cc from "" to "jtatar, cweaver, sflis, tfuchs, claudio.kopper, hosina, jbraun, jvansanten"
* olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> PROPOSAL
> clsim
> credo
> lilliput
> millipede
> neutrino-generator
> paraboloid
> photonics-service
> photospline
> sim-services
> wavereform
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> clsim
> credo
> lilliput
> millipede
> neutrino-generator
> paraboloid
> photonics-service
> photospline
> sim-services
> wavereform
> 


* olivas changed owner from "olivas, jtatar, cweaver, sflis, tfuchs,  claudio.kopper, hoshina, jbraun, jvansanten" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20150812T23:55:47: olivas changed cc from "jtatar, cweaver, sflis, tfuchs, claudio.kopper, hosina, jbraun, jvansanten" to "jtatar, cweaver, sflis, claudio.kopper, hosina, jbraun, jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20150812T23:56:18: olivas changed cc from "jtatar, cweaver, sflis, claudio.kopper, hosina, jbraun, jvansanten" to "jtatar, cweaver, sflis, claudio.kopper, hoshina, jbraun, jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20150908T19:48:18: olivas changed status from "assigned" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20160213T15:07:39: 
* claudio.kopper commented:

> clsim does only have a spicy dependence in a maintenance/plotting script not used in tests not in production. I would say this is not a hard dependence and should not be listed in CMakeLists.txt. I removed the dependency that was added for clsim and removed myself from this ticket.


* claudio.kopper changed cc from "jtatar, cweaver, sflis, claudio.kopper, hoshina, jbraun, jvansanten" to "jtatar, cweaver, sflis, hoshina, jbraun, jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20160218T00:24:15: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> clsim
> credo
> lilliput
> millipede
> neutrino-generator
> paraboloid
> photonics-service
> photospline
> sim-services
> wavereform
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> wavereform
>

**Comment by icecube-trac** [comment]
Trac update at 20160218T00:33:12: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> DOMLauncher
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> wavereform
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> wavereform
>

**Comment by icecube-trac** [comment]
Trac update at 20160218T00:36:59: 
* olivas changed owner from "olivas" to "jtatar"
* olivas changed status from "accepted" to "assigned"
* olivas commented:

> Reassigning to Joulien who's next on the list with credo.  How about we just shuffle through the maintainers likes this.  Joulien can pass it on to Kai (owner of lilliput), and so on...


* olivas changed cc from "jtatar, cweaver, sflis, hoshina, jbraun, jvansanten" to "jtatar, jbraun, jvansanten, kkrings, kjmeagher"

**Comment by icecube-trac** [comment]
Trac update at 20160219T17:23:41: kjmeagher commented:

> In changeset 142125:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="142125"
> paraboloid only depends on scipy for pyraboloid, which is not currently used by anything and is not in any test suite see #1104
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160219T20:27:59: jbraun commented:

> wavereform no longer depends on scipy. I've removed the unnecessary scipy dependence from CMakeLists.txt.

**Comment by icecube-trac** [comment]
Trac update at 20160219T20:28:15: jbraun changed cc from "jtatar, jbraun, jvansanten, kkrings, kjmeagher" to "jtatar, jvansanten, kkrings, kjmeagher"

**Comment by icecube-trac** [comment]
Trac update at 20160225T17:07:05: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> wavereform
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160225T17:11:56: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> paraboloid
> photonics-service
> photospline
> ~~sim-services~~
> ~~wavereform~~
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service
> photospline
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160225T17:15:41: 
* olivas commented:

> Added Jim back on the cc list so he can take a look at millipede.


* olivas changed cc from "jtatar, jvansanten, kkrings, kjmeagher" to "jtatar, jvansanten, kkrings, jbraun"
* olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo
> lilliput
> millipede
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service
> photospline
> ~~sim-services~~
> ~~wavereform~~
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> lilliput (Kai)
> millipede (Jim)
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160225T17:33:09: 
* jbraun commented:

> scipy is only needed in Millipede for the Millipede artist.  Since the build/tests can be done without scipy, I've removed the dependence.


* jbraun changed cc from "jtatar, jvansanten, kkrings, jbraun" to "jtatar, jvansanten, kkrings"

**Comment by icecube-trac** [comment]
Trac update at 20160225T17:34:54: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> lilliput (Kai)
> millipede (Jim)
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> lilliput (Kai)
> ~~millipede~~
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160304T08:25:22: kkrings commented:

> Only Lilliput's SciPyMinimizer depends on SciPy and this minimizer is more like an optional feature. So, I have removed the dependency in CMakeLists.txt. The corresponding test is omitted if the minimizer cannot be imported.

**Comment by icecube-trac** [comment]
Trac update at 20160311T23:49:38: olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> lilliput (Kai)
> ~~millipede~~
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> ~~lilliput~~
> ~~millipede~~
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160328T17:01:52: 
* olivas changed owner from "jtatar" to "jvansanten"
* olivas changed cc from "jtatar, jvansanten, kkrings" to ""
* olivas commented:

> credo only uses scipy for diagnostics so is not a hard dependency.


* olivas changed description from:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> credo (Joulien)
> ~~lilliput~~
> ~~millipede~~
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
> 

to:

> Determine whether scipy is really a hard dependence or just needed for supplemental features.  If it's not a hard dependence disable whatever depends on scipy in CMakeLists.txt
> 
> ~~DOMLauncher~~ (only two examples use scipy)
> ~~clsim~~
> ~~credo~~
> ~~lilliput~~
> ~~millipede~~
> ~~neutrino-generator~~
> ~~paraboloid~~
> photonics-service (Jakob)
> photospline (Jakob)
> ~~sim-services~~
> ~~wavereform~~
>

**Comment by icecube-trac** [comment]
Trac update at 20160401T08:04:18: jvansanten commented:

> In changeset 144112:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="144112"
> Remove scipy dependency, as it is used only in the Python reference
> implementation and is protected in a try/except ImportError block. See #1104.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160401T08:06:07: 
* jvansanten changed status from "assigned" to "closed"
* jvansanten commented:

> In changeset 144113:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="144113"
> Remove scipy dependency, as it is only used in a demo script. Closes #1104
> (finally!)
> ```


* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:57: 
* olivas changed milestone from "IceRec V5" to ""
* olivas commented:

> Milestone IceRec V5 deleted
